File: Javascript/bubbleSort.js
function bubbleSort(arr) {
    let i, j;
    let len = arr.length;

    let isSwapped = false;

    for (i = 0; i < len; i++) {
        isSwapped = false;

        for (j = 0; j < len; j++) {
            if (arr[j] > arr[j + 1]) {
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                isSwapped = true;
            }
        }

        if (!isSwapped) {
            break;
        }
    }
  console.log(arr);
}

let arr = [243, 45, 23, 356, 3, 5346, 35, 5];

bubbleSort(arr);

File: DSA/DSA-Java/Sorting-Algorithms/BubbleSort.java
class BubbleSort {
    public static void main(String[] args) {
        int [] arr = {2, 5, 1, 7};
        sort(arr);
        for (int el: arr) {
            System.out.print(el + " ");
        }
    }

    public static void sort(int [] array) {
        boolean sorted = false;
        int temp;
        while(!sorted) {
            sorted = true;
            for (int i = 0; i < array.length - 1; i++) {
                if (array[i] > array[i+1]) {
                    temp = array[i];
                    array[i] = array[i+1];
                    array[i+1] = temp;
                    sorted = false;
                }
            }
        }
    }
}